src/services/databaseObjectSearchService.ts

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import SqlToolsServiceClient from "../languageservice/serviceclient";
+import { ObjectMetadata, SimpleExecuteResult } from "vscode-mssql";
+import { RequestType } from "vscode-languageclient";
+import { getErrorMessage } from "../utils/utils";
+
+export interface DatabaseObject {
+    name: string;
+    type: string;
+    schema: string;
+    fullName: string;
+}
+
+export interface SearchResult {
+    success: boolean;
+    objects: DatabaseObject[];
+    error?: string;
+}
+
+export class DatabaseObjectSearchService {
+    private _client: SqlToolsServiceClient;
+    private _metadataCache: Map<string, ObjectMetadata[]> = new Map();
+
+    constructor(client?: SqlToolsServiceClient) {
+        this._client = client || SqlToolsServiceClient.instance;
+    }
+
+    /**
+     * Search for database objects matching the given search term
+     * @param connectionUri The URI of the active connection
+     * @param searchTerm The search term to look for
+     * @param database The name of the database to search in. If not provided, the search is performed on the default database for the connection.
+     * @returns A promise that resolves to the search results
+     */
+    public async searchObjects(
```

Comment:
Please switch to metadata/list apis.  It takes care of query parameterization and escaping. 

```
export interface MetadataListResult {
    metadata: ObjectMetadata[];
}

export namespace MetadataListRequest {
    export const type = new RequestType<MetadataListParams, MetadataListResult, void, void>(
        "metadata/list",
    );
}
const results = await this._client.sendRequest(MetadataListRequest.type, {
    ownerUri: connectionUri,
});
```